styx-api-service/src/main/java/com/spotify/styx/api/Middlewares.java

```diff
@@ -37,6 +44,32 @@
  */
 public final class Middlewares {
 
+  private static class ClientValidator implements
+                               Middleware<AsyncHandler<? extends Response<?>>,
+                                   AsyncHandler<? extends Response<ByteString>>> {
+    private final Supplier<List<String>> clientBlacklist;
+
+    ClientValidator(Storage storage) {
+      clientBlacklist = new CachedSupplier<>(storage::clientBlacklist, Instant::now);
```

Comment:
I would rather pass in a `Supplier<List<String>>` as a dependency instead of constructing it from `Storage` here.

This should also use the injected `Time` instance instead of using `Instant::now`. But this should be solved by just moving the construction up to the main service class.